toolchain/check/impl.cpp

```diff
@@ -255,6 +255,7 @@ auto FillImplWitnessWithErrors(Context& context, SemIR::Impl& impl) -> void {
         elem = SemIR::ErrorInst::InstId;
       }
     }
+    impl.witness_id = SemIR::ErrorInst::InstId;
```

Comment:
Should the `impl.witness_id` be set to `Error` when it's `None`?

Note, the outer `if` could also use an early return to reduce the indent. :)

Reply:
It should be yeah.